packages/@aws-cdk/aws-cloudwatch/test/insight-rule.test.ts

```diff
@@ -0,0 +1,40 @@
+import { Template } from '@aws-cdk/assertions';
+import { Stack } from '../../core/lib/stack';
+import { AggregateOptions, InsightRule, LogFormat } from './../lib/insight-rule';
+
+describe('Insight Rule', () => {
+  let logGroupNames = ['<loggroupname>'];
+  let ruleName = 'my-insight-rule';
+  let keys = [
+    '$.key_1',
+    '$.key_2',
+  ];
+
+  test('can create an insight rule', () => {
+    const stack = new Stack();
+
+    new InsightRule(stack, 'MyInsightRule', {
+      logGroupNames,
+      insightRuleName: ruleName,
+      keys,
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::CloudWatch::InsightRule', {
+      RuleBody: JSON.stringify({
+        AggregateOn: AggregateOptions.COUNT,
+        Contribution: {
+          Keys: keys,
+          Filters: [],
```

Comment:
Would there be value in testing the inclusion of various filters?

Reply:
+1. Whatever we decide for the design of the `RuleBody` property, we are going to have to thoroughly unit test every possible outcome. The current unit test is a good start, but we will need far more.